### PR TITLE
Delete partners.PCA models and friends

### PR DESCRIPTION
**Pull request by caktus-philip**
Delete the following partners models --
 - `AmendmentLog`
 - `AuthorizedOfficer`
 - `BankDetails`
 - `GwPCALocation`
 - `IndicatorDueDates`
 - `PartnershipBudget`
 - `PCA`
 - `PCAFile`
 - `PCAGrant`
 - `PCASector`
 - `PCASectorGoal`
 - `RAMIndicator`

Also delete lots of related serializers, filters, forms, etc.

Add `partners.models.AgreementStatus` to preserve constants that had been in the `PCA` model and are still in use.

**Question 1** -- the following entities changed substantially as a result of the models above being removed. Should I remove them also?
 - `partners.views.v1.AgreementViewSet`
 - `partners.serializers.v1.LocationSerializer`
 - `partners.serializers.v1.PartnerOrganizationSerializer`

**Question 2** -- `reports.Goal` doesn't seem to be used anymore, should I remove it?

**Question 3** -- Did I remove everything that should be removed? Maybe I removed 
too much?

### CODE REVIEW

File: EquiTrack/partners/models.py [hunk 1]
```diff
@@ -21,28 +21,45 @@
...
 from users.models import Section, Office
-from notification.models import Notification
 from partners.validation.agreements import (
     agreement_transition_to_ended_valid,
     agreements_illegal_transition,
     agreement_transition_to_signed_valid)
 from partners.validation import interventions as intervention_validation
 
 
+class AgreementStatus(object):
```
Comment (robertavram) [inline EquiTrack/partners/models.py]: this should not be used anywhere important.. probably can be removed

File: EquiTrack/partners/models.py [hunk 2]
```diff
@@ -1984,731 +1970,12 @@ class DirectCashTransfer(models.Model):
...
-        super(PartnershipBudget, self).save(**kwargs)
-
-    def __unicode__(self):
-        return '{}: {}'.format(
-            self.partnership,
-            self.total
-        )
-
-
 class AgreementAmendmentLog(TimeStampedModel):
```
Comment (robertavram) [inline EquiTrack/partners/models.py]: this whole model can / should be removed please :)

File: EquiTrack/reports/models.py [hunk 3]
```diff
@@ -448,9 +448,9 @@ def __str__(self):
         )
 
     def programmed_amounts(self):
```
Comment (robertavram) [inline EquiTrack/reports/models.py]: this function should be removed.. can you check where it's used also?

**Comment by robertavram** [comment]
Question 2 -- reports.Goal doesn't seem to be used anymore, should I remove it?
> YES. :)

**Comment by robertavram** [comment]
ok.. so also remove these:
PartnerStaffMemberPropertiesAPIView
PartnerStaffMemberPropertiesSerializer
AgreementViewSet
IndicatorReportViewSet
PartnerOrganizationsViewSet
PartnerStaffMembersViewSet

Serializers:
RapidProRequest
PartnershipExportFilterSerializer
AgreementExportFilterSerializer

this should clean things up a bit more

**Comment by caktus-philip** [comment]
@robertavram Thanks for the review. 


I deleted these models --
 - `partners.models.AgreementAmendmentLog`
 - `partners.models.IndicatorReport`
 - `reports.models.Goal`

And these views --
 - `partners.views.v1.AgreementViewSet`
 - `partners.views.v1.PartnerOrganizationsViewSet`
 - `partners.views.v1.PartnerStaffMembersViewSet`
 - `partners.views.v1.IndicatorReportViewSet`
 - `partners.views.v2.PartnerStaffMemberPropertiesAPIView`

And these serializers -- 
 - `reports.serializers.v1.SectorSerializer (renamed sibling serializer SectorLightSerializer to SectorSerializer)`
 - `partners.serializers.partner_organization_v2.PartnerStaffMemberPropertiesSerializer`
 - `partners.serializers.v1.AgreementExportFilterSerializer`
 - `partners.serializers.v1.PartnerStaffMemberSerializer`
 - `partners.serializers.v1.PartnershipExportFilterSerializer`
 - `partners.serializers.v1.RapidProRequest`
 - `partners.serializers.v1.IndicatorReportSerializer`

And these filters and permissions -- 
 - `partners.permissions.PartnerPermission`
 - `partners.filters.PartnerOrganizationExportFilter`
 - `partners.filters.AgreementExportFilter`

**Question**
Should I also remove `PartnerStaffMemberPropertiesView`? Should I just remove everything that remains in `partners/urls.py`?

**Caveats**

 - `IndicatorReport` is still mentioned in two front end files -- 
https://github.com/unicef/etools/blob/develop/EquiTrack/assets/frontend/partner/elements/partner_elements.vulcanized.html
https://github.com/unicef/etools/blob/develop/EquiTrack/assets/frontend/partner/scripts/app.js

 - `reports.Indicator.programmed_amounts()` is still in use. It's used by `Indicator` model methods `programmed()` and `progress()`. As far as I can tell, `progress()` isn't used, but `programmed()` is used in these two templates --
 - https://github.com/unicef/etools/blob/develop/EquiTrack/templates/dashboard.html
 - https://github.com/unicef/etools/blob/develop/EquiTrack/templates/map.html

Should those templates maybe use `indicator.total` instead? I'm just guessing.

**Comment by ntrncic** [comment]
Can remove `PartnerStaffMemberPropertiesView`
Can remove `IndicatorReport` and can remove all the template files that reference it.

**Comment by caktus-philip** [comment]
@robertavram 

I think the last set of references to clean up are all related to `reports.Indicator.programmed_amounts()`. That method is the only thing referencing `AgreementStatus`. It's also referencing `self.resultchain_set` which I don't think exists, so the method may be entirely broken. If we can figure out what the two templates below should use instead of `reports.Indicator.programmed()`, I can remove `reports.Indicator.programmed_amounts()/.progress()/.programmed()` and `AgreementStatus` and then this PR might be ready.

 - https://github.com/unicef/etools/blob/develop/EquiTrack/templates/dashboard.html
 - https://github.com/unicef/etools/blob/develop/EquiTrack/templates/map.html

Should those templates maybe use `indicator.total` instead? I'm just guessing.

File: EquiTrack/partners/models.py (hunk 1 above)
Comment (robertavram) [reply EquiTrack/partners/models.py]: @caktus-philip can we please :) remove this?

**Pull request merged**